src/core/reference/include/openvino/reference/non_zero.hpp

```diff
@@ -17,26 +20,16 @@ namespace reference {
 /// Output number of non-zero entries in arg
 template <typename T>
 size_t non_zero_get_count(const T* arg, const Shape& arg_shape) {
-    T zero = 0;
-    size_t arg_rank = arg_shape.size();
-    size_t arg_count = shape_size(arg_shape);
-    size_t non_zero_count = 0;
-
-    // Input arg is scalar
-    if (arg_rank == 0) {
-        if (*arg != zero) {
-            non_zero_count = 1;
-        }
-    } else  // Input is non scalar case
-    {
-        for (size_t i = 0; i < arg_count; i++) {
-            if (arg[i] != zero) {
-                non_zero_count++;
-            }
-        }
+    const auto zero = T{0};
+    const auto arg_count = shape_size(arg_shape);
+    switch (arg_count) {
+    case 0:
+        return 0;
+    case 1:
+        return *arg != zero;
+    default:
+        return arg_count - std::count(arg, arg + arg_count, zero);
     }
```

Comment:
```suggestion
    auto arg_count = shape_size(arg_shape);
    if (arg_cout > 0){
        arg_count -= std::count(arg, arg + arg_count, zero);
    }
    return arg_count;
```
There is not much benefit for multi-return for fundamental type. Just alternative to consider.

Reply:
done